slsReceiverSoftware/src/DataProcessor.cpp

```diff
@@ -519,62 +579,216 @@ void DataProcessor::PadMissingPackets(sls_receiver_header header, char *data) {
     }
 }
 
-/** ctb specific */
-void DataProcessor::RearrangeDbitData(size_t &size, char *data) {
-    int nAnalogDataBytes = generalData->GetNumberOfAnalogDatabytes();
-    int nDigitalDataBytes = generalData->GetNumberOfDigitalDatabytes();
-    int nTransceiverDataBytes = generalData->GetNumberOfTransceiverDatabytes();
-    // TODO! (Erik) Refactor and add tests
-    int ctbDigitalDataBytes = nDigitalDataBytes - ctbDbitOffset;
+void DataProcessor::RemoveTrailingBits(size_t &size, char *data) {
+    const size_t nAnalogDataBytes = generalData->GetNumberOfAnalogDatabytes();
+    const size_t nDigitalDataBytes = generalData->GetNumberOfDigitalDatabytes();
+    const size_t nTransceiverDataBytes =
+        generalData->GetNumberOfTransceiverDatabytes();
+
+    const size_t ctbDigitalDataBytes = nDigitalDataBytes - ctbDbitOffset;
 
     // no digital data
     if (ctbDigitalDataBytes == 0) {
         LOG(logWARNING)
-            << "No digital data for call back, yet dbitlist is not empty.";
+            << "No digital data for call back, yet ctbDbitOffset is non zero.";
         return;
     }
 
-    const int numDigitalSamples = (ctbDigitalDataBytes / sizeof(uint64_t));
+    // update size and copy data
+    memmove(data + nAnalogDataBytes, data + nAnalogDataBytes + ctbDbitOffset,
```

Comment:
memmove copied the data to a temporary array first. (https://en.cppreference.com/w/cpp/string/byte/memmove) do we need this or could we copy directly to the destination?